manifests/init.pp

```diff
@@ -46,22 +46,16 @@
 # Sample Usage:
 #
 class java(
-  $distribution          = 'jdk',
-  $version               = 'present',
-  $package               = undef,
-  $package_options       = undef,
-  $java_alternative      = undef,
-  $java_alternative_path = undef,
-  $java_home             = undef
+  String $distribution                                              = 'jdk',
+  Pattern[/present|installed|latest|^[.+_0-9a-zA-Z:~-]+$/] $version = 'present',
+  $package                                                          = undef,
+  Optional[Array] $package_options                                  = undef,
+  $java_alternative                                                 = undef,
```

Comment:
The goal was to replace all the validate_* calls with datatypes, not to add datatypes to all parameters. I normally mention this, but somehow forgot it here. I added datatypes to all params in init.pp now.